torchvision/prototype/transforms/functional/_augment.py

```diff
@@ -19,7 +19,15 @@ def erase_image_pil(
     return to_pil_image(output, mode=img.mode)
 
 
-def erase(inpt: Any, i: int, j: int, h: int, w: int, v: torch.Tensor, inplace: bool = False) -> Any:
+def erase(
+    inpt: Union[torch.Tensor, PIL.Image.Image, features.Image],
```

Comment:
Shouldn't we also check inside `F.erase` for invalid types like BBoxes, Masks, Labels explicitly ?
As it is a public functional API, this code can be used without Transforms and thus can have a similar issue trying erase boxes...

Reply:
Right now, the new `F` follows the same idiom/approach as the old `F`. It checks for specific types in a series of if/elif and the else redirects to the default kernel (whichever this is). This can be error prone, I agree. What we could do is send a separate refactoring PR that changes the idiom on the new `F` to throw an exception on the `else` indicating that the input is not recognised. I wouldn't explicitly look for things we don't support but instead check for those that we do and throw the error on the `else` statement.